src/EditorFeatures/Core/ExtractInterface/AbstractExtractInterfaceCommandHandler.cs

```diff
@@ -78,6 +78,7 @@ public bool ExecuteCommand(ExtractInterfaceCommandArgs args, CommandExecutionCon
                         return;
                     }
 
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
can also do .ConfigureAwait(true) on ExtractInterfaceAsync above, with a comment that this will bring us back to the UI thread for TryApplyChanges.

Reply:
above would be my preference, but either approach is fine to me.